vmm/src/lib.rs

```diff
@@ -782,8 +790,20 @@ impl Vmm {
                 .insert(cfg.iface_id.clone(), handler_idx);
 
             let allow_mmds_requests = cfg.allow_mmds_requests();
-            let rx_rate_limiter = cfg.rx_rate_limiter.take();
-            let tx_rate_limiter = cfg.tx_rate_limiter.take();
+            let rx_rate_limiter = match cfg.rx_rate_limiter {
```

Comment:
nit: you could do this and the following 4 instances of the same operation using the rusty [.map()](https://doc.rust-lang.org/std/option/enum.Option.html#method.map)

Reply:
It would look nicer, indeed. However, since `into_rate_limiter()` may fail, I couldn't find a way to propagate the resulted error out from the closure used by `map()`. I.e.
```rust
let rx_rate_limiter = cfg.rx_rate_limiter.map(|rl| rl.into_rate_limiter()?)
```
doesn't work, because the closure cannot return a `Result` object.